tests/functional/test_fusion_volume.py

```diff
@@ -0,0 +1,676 @@
+    fusion_volume,
+)
+from ansible_collections.purestorage.fusion.tests.functional.utils import (
+    AnsibleExitJson,
+    AnsibleFailJson,
+    OperationMock,
+    SuccessfulOperationMock,
+    exit_json,
+    fail_json,
+    set_module_args,
+)
+from urllib3.exceptions import HTTPError
+
+# GLOBAL MOCKS
+fusion_volume.setup_fusion = MagicMock(return_value=purefusion.api_client.ApiClient())
+purefusion.api_client.ApiClient.call_api = MagicMock(
+    side_effect=Exception("API call not mocked!")
+)
+basic.AnsibleModule.exit_json = exit_json
+basic.AnsibleModule.fail_json = fail_json
+
+
+@pytest.fixture
+def module_args():
+    return {
+        "name": "volume_1",
+        "state": "present",
+        "display_name": "Volume 1",
+        "tenant": "t1",
+        "tenant_space": "ts1",
+        "placement_group": "pg1",
+        "storage_class": "sc1",
+        "protection_policy": "pp1",
+        "host_access_policies": ["hap1"],
+        "eradicate": False,
+        "size": "1M",
+        "issuer_id": "ABCD1234",
+        "private_key_file": "private-key.pem",
+    }
+
+
+@pytest.fixture
+def absent_module_args(module_args):
+    module_args.update(
+        {"host_access_policies": [], "eradicate": True, "state": "absent"}
+    )
+    return module_args
+
+
+@pytest.fixture
+def storage_class():
+    return {
+        "name": "sc1",
+        "display_name": "Storage Class 1",
+        "storage_service": "storage_service_1",
+        "size_limit": 4096,
+        "iops_limit": 100,
+        "bandwidth_limit": 1024,
+        "self_link": "self_link",
+        "id": "id_1",
+    }
+
+
+@pytest.fixture
+def placement_group():
+    return {
+        "name": "pg1",
+        "display_name": "pg1",
+        "tenant": "t1",
+        "tenant_space": "ts1",
+        "availability_zone": "az1",
+        "storage_service": "ss1",
+        "self_link": "self_link",
+        "id": "id_1",
+    }
+
+
+@pytest.fixture
+def protection_policy():
+    return {
+        "name": "pp1",
+        "display_name": "pp1",
+        "self_link": "self_link",
+        "id": "id_1",
+        "objectives": "objectives",
+    }
+
+
+@pytest.fixture
+def host_access_policies():
+    return {
+        "name": "hap1",
+        "display_name": "hap1",
+        "iqn": "iqn1",
+        "personality": "linux",
+        "self_link": "self_link",
+        "id": "id_1",
+    }
+
+
+@pytest.fixture
+def volume(storage_class, placement_group, protection_policy, host_access_policies):
+    return {
+        "name": "volume_1",
+        "display_name": "Volume 1",
+        "tenant": "t1",
+        "tenant_space": "ts1",
+        "storage_class": purefusion.StorageClass(**storage_class),
+        "placement_group": purefusion.PlacementGroup(**placement_group),
+        "protection_policy": purefusion.ProtectionPolicy(**protection_policy),
+        "host_access_policies": [purefusion.HostAccessPolicy(**host_access_policies)],
```

Comment:
No need to add actual objects. In fact, if you want to use correct values, you should use `StorageClassRef`, `PlacementGroupRef` etc. [Here](https://github.com/PureStorage-OpenConnect/fusion-python-sdk/blob/main/fusion/models/volume.py) you can check actual parameters of the Volume object.

> swagger_types = {
        'size': 'int',
        'tenant': 'TenantRef',
        'tenant_space': 'TenantSpaceRef',
        'storage_class': 'StorageClassRef',
        'protection_policy': 'ProtectionPolicyRef',
        'placement_group': 'PlacementGroupRef',
        'array': 'ArrayRef',
        'created_at': 'int',
        'source_volume_snapshot': 'VolumeSnapshotRef',
        'host_access_policies': 'list[HostAccessPolicyRef]',
        'serial_number': 'str',
        'target': 'Target',
        'time_remaining': 'int',
        'destroyed': 'bool'
    }

I suggest using mocked string value like `"this_is_mock"` or something similar so you don't have hundreds of lines here.